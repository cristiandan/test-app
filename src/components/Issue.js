import React from 'react';

export default ({ title }) => {
    return <p>{title}</p>;
};
