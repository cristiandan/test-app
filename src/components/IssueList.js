import React from 'react';
import { connect } from 'react-redux';
import Issue from './Issue';

const IssueList = ({ repoName, issues = [], issuesLoading = [] }) => {
    const issuesList = issues[repoName]
        ? issues[repoName].map(item => <Issue {...item} key={item.id} />)
        : '';
    return (
        <div>
            {issuesLoading[repoName] && <div>Loading</div>}
            {issuesList}
        </div>
    );
};

export default connect(state => ({
    issues: state.issues,
    issuesLoading: state.issuesLoading,
}))(IssueList);
