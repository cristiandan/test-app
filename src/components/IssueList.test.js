import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { IssueList } from './IssueList';
import Issue from './Issue';

const props = {
    repoName: 'Repo 1',
    issues: {
        'Repo 1': [{ id: 1, title: 'Issue1' }, { id: 3, title: 'Issue2' }],
    },
    issuesLoading: { 'Repo 1': false },
};

test('IssueList should render correctly with props', () => {
    const wrapper = mount(<IssueList {...props} />);
    expect(toJson(wrapper.render())).toMatchSnapshot();
    expect(wrapper.find(Issue).first().text()).toEqual('Issue1');
    expect(wrapper.find(Issue).last().text()).toEqual('Issue2');
});

const propsLoading = {
    repoName: 'Repo 1',
    issues: {
        'Repo 1': [],
    },
    issuesLoading: { 'Repo 1': true },
};

test('IssueList should render correctly on loading', () => {
    const wrapper = mount(<IssueList {...propsLoading} />);
    expect(toJson(wrapper.render())).toMatchSnapshot();
    expect(wrapper.find('div').last().text()).toEqual('Loading');
});

test('IssueList should render without props', () => {
    const wrapper = mount(<IssueList {...propsLoading} />);
    expect(toJson(wrapper.render())).toMatchSnapshot();
});
