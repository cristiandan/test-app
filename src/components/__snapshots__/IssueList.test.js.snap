// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueList should render correctly on loading 1`] = `
<div>
  <div>
    Loading
  </div>
</div>
`;

exports[`IssueList should render correctly with props 1`] = `
<div>
  <p>
    Issue1
  </p>
  <p>
    Issue2
  </p>
</div>
`;

exports[`IssueList should render without props 1`] = `
<div>
  <div>
    Loading
  </div>
</div>
`;
